components/higher-order/wrapper-display-name/index.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+
+export default function wrapperDisplayName( wrapperPrefix, WrappedComponent ) {
```

Comment:
This is not really a higher-order component, since the return value from the function is a string, not the enhanced component. I think we'd want to do one of either:

- Move this elsewhere to better reflect that it is not a higher-order component
- Make it in-fact a higher-order component, i.e. return an enhanced component with `displayName` assigned

My preference is toward the latter.

Reply:
True, it’s only related logic. It’s not HOC. 

Just to confirm. Option 2 would be a HOC called inside another HOC that would update displaName, right?

Reply:
>Just to confirm. Option 2 would be a HOC called inside another HOC that would update displaName, right?

Yes. Though now that I'm thinking about it, layering components like this risks being a performance hazard ([reference](https://github.com/acdlite/recompose/blob/master/docs/performance.md)).

Reply:
Unless we do perf optimization and structure like a HOC but behind the scenes only update displayName on the component:

const withDisplayName = ( prefix, SourceComponent ) => ( WrappedComponent ) => {
    WrappedComponent.displayName = ...;
    return WrappedComponent;
}

Reply:
After some thinking, I'd rather move it to `@wordpress/utils` to `components.js` file and use it as it is.

Reply:
Would it be a better fit in `@wordpress/element` ? `utils` as a dumping ground is not scalable.

Reply:
Moved and documented.